### ISSUE TITLE
video playback performance issues with build 1.0.20 ????

### ISSUE THREAD

**Issue by DavidXanatos** [sandboxie-plus/Sandboxie#1841]
There were some reports about decreased video playback performance with the build 1.0.20
I'm currently unable to replicate those, hence i would ask you to test this out and report back how the version works for you.

Things to test
1. versions prior to 1.0.18
2. adding ReadIpcPath=$:* to sandboxie ini

it important to know if the issue was introduced with 1.0.18 and or of its related to the new memory isolation.

**Comment by mysteriously** [comment]
I have Dell E5450 (i5-5300u + Nvidia 830M). I have never been able to play 3 full screen 1080p@25fps videos smoothly in Firefox (one for each display). However using 1.0.20 I am not able to play 720p and 360p

**Comment by DavidXanatos** [comment]
does using 1.0.18 fixes this? and or how dows 1.0.19 work?
also if that does not help does the ReadIpcPath=$:*  help?
and if not whats the last working version?

**Comment by mysteriously** [comment]
firefox.exe is the only one allowed to run. I will do those tests and let you know

**Comment by Rngexile** [comment]
Cpu intel i7 3770k here.
Win10 21H1 x64

**Tested with sandboxie-plus 1.0.19 : NO video playback performance issues**
- firefox (100.0) with 3-6 1080p videos at the same time on youtube (no drm, no plugin-contrainer.exe involved)
- firefox (100.0) with 3-6 1080p videos at the same time on various websites (animes, dailymotion and so on) (no drm, no plugin-contrainer.exe involved)
- +20Go h264 personnal video files on HDD : 3-4 at the same time with video players mpc-hc and vlc (3.017.4)

**Tested with sandboxie-plus 1.0.20 : NO video playback performance issues (similar %cpu/gpu used as 1.0.19, and same general behaviour, all is fine)**
- firefox (100.0) with 3-6 1080p videos at the same time on youtube (no drm, no plugin-contrainer.exe involved)
- firefox (100.0) with 3-6 1080p videos at the same time on various websites (animes, dailymotion and so on) (no drm, no plugin-contrainer.exe involved)
**- firefox (100.0) with 3-6 1080p videos at the same time on https://www.viki.com/v1/explore?access=free (maybe with drm,  plugin-contrainer.exe involved with low cpu use 3.5-6.5% and no gpu use)**
- +20Go h264 personnal video files on HDD : 3-4 at the same time with video players mpc-hc and vlc (3.017.4)

I can't reproduce any video playback performance issues with sandboxie-plus 1.0.20 (or below) on my side and with my use (no netflix and so on).

**Comment by mysteriously** [comment]
Yes, 1.0.18 does not have performance playback issue. However it has other issue - sometimes videos take very long time to load and firefox tries to use software acceleration.

**Comment by DavidXanatos** [comment]
can you please try builds 1.0.14 and 1.0.15

**Comment by DavidXanatos** [comment]
Build 1.0.21 is out, it does not fixed the playback issue some users seam to have, but it adds a new powerful debug facility that can help pin down the problem: '''FuncSkipHook=FunctionName''' allows selectively to disable individual sandboxie hooker per function.

please try adding 
```FuncSkipHook=NtOpenSymbolicLinkObject```
to your sandboxie ini and please report back if that improves the playback performance.
please note that this will break compatibility with mingw and possibly other things.

If it, as expected, helps than the next step will be to record trace logs in "monitor mode" with and without the setting, sort them booth by name save to two files and compare. using some compare tool, you can also post booth logs here and I compare them for you.

**Comment by DavidXanatos** [comment]
@mysteriously had you time to test 1.0.21 with the aformentioned setting option?

**Comment by mysteriously** [comment]
Yes. Without it it was lagging when playing multiple videos. After I added it, it was fine.
I hope this helps. If not, maybe we should use the same websites to generate load
[Logs.zip](https://github.com/sandboxie-plus/Sandboxie/files/8676319/Logs.zip)

**Comment by DavidXanatos** [comment]
@mysteriously please re record the logs in monitor mode
![grafik](https://user-images.githubusercontent.com/3890945/168045362-38e1e5f4-3354-473c-89a9-830a4b3084cb.png)
this button must be pressed

**Comment by mysteriously** [comment]
When using Dark theme, the monitor icon is the same regardless if it is pressed or not. I thought it preserves the log and switches just the view.
Also, it is not possible to save the log when monitor mode is enabled. I will copy the content of the window

**Comment by mysteriously** [comment]
[Logs2.zip](https://github.com/sandboxie-plus/Sandboxie/files/8677504/Logs2.zip)
There you go. BTW: For some reason 1.0.21 is not that slow like 1.0.20

**Comment by DavidXanatos** [comment]
So to clarify 
whoch of teh 2 files is with the FuncSkipHook=NtOpenSymbolicLinkObject option?

**Comment by mysteriously** [comment]
omg, I am so sorry. WithoutHook is in fact a log without the magic line you told me to add

**Comment by DavidXanatos** [comment]
yea i suspected that much, so the next step will be to test the ReadIpcPath i posted above

**Comment by mysteriously** [comment]
Yes, with all of those added, everything worked fine. How does it look like for others?

**Comment by isaak654** [comment]
> Yes, with all of those added, everything worked fine.

That's great!
Please try removing gradually the lines and repeating the test until only those necessary for smooth playback remain.

**Comment by DavidXanatos** [comment]
@mysteriously it would be great if you coudl do the testing soon, i woudl like to release a 1.0.22 build for monday with the fix

**Comment by mysteriously** [comment]
Do you want me to test from top to down? With 10 entries there are insane numbers of combinations.
Everything with fine with commented out ReadIpcPath=\Device{*

**Comment by DavidXanatos** [comment]
The order does not mater, 
so the fastest route to a result is to comment out half of the entries,
and test if that still works,
if it does, great comment out the next half of the remaining half and son on
if it does not uncomment half of what you previously commented out, if that works great if still not the other half.
That strategy assumes that its one entry that's needed.

this may not necessarily be true, so an other aproche would be to comment out lines one by one and testing each time, uncommenting lines that are nececery such that in the end only required lines are left.

**Comment by mysteriously** [comment]
I commented out all of them and was able to play 3 720p videos smoothly. So it does not seem to be so easy to reproduce with 1.0.21

**Comment by DavidXanatos** [comment]
This should not be possible, build 1.0.21 does nto have any changes with regard to that.
Anyhow you can use ReadIpcPath with 1.0.20 hence please test with the previous build

**Comment by DavidXanatos** [comment]
@mysteriously so how does it look?

**Comment by mysteriously** [comment]
I can not reproduce it anymore. When it occurs, I am not able to play 720p and 480p videos at the same time. Now I can 3x720p. With 1.0.21 seems slightly less laggy than 1.0.21.
The only changes on my systems were windows updates, Firefox update and Steam update

You mentioned other people also have this issue but I can't find anything here

**Comment by DavidXanatos** [comment]
It was one other user but he cant reproduce it eider so idk... may be there never was a performance bug

**Comment by mysteriously** [comment]
Well, there was something for sure, as I was not able to play multiple videos for few days. It started 10 days ago as mentioned https://github.com/sandboxie-plus/Sandboxie/issues/1831#issuecomment-1118353192
I will keep you posted if I find anything interesting

**Comment by bastik-1001** [comment]
I reported the issue with encrypted videos not being played, but I have not noticed any performance issues regarding video playback.

**Comment by mysteriously** [comment]
I cleaned up history in the sandbox. For some reason all my tabs from previous session were gone. After deleting sandbox content I was able to reproduce the issue. The only recent change on my PC were some windows updates. I will check it out outside of sandboxie and with various browsers.

In fact I should use Windows 10 on this garbage laptop, as it is last supported OS

**Comment by mysteriously** [comment]
Yesterday the issue occurred with v1.0.18

**Comment by isaak654** [comment]
#3102
